domain/src/main/java/org/oppia/android/domain/survey/SurveyQuestionDeck.kt

```diff
@@ -91,4 +93,14 @@ class SurveyQuestionDeck constructor(
   private fun isTopOfDeckTerminal(): Boolean {
     return isTopOfDeckTerminalChecker(pendingTopQuestion)
   }
+
+  /** Stores a list of all the questions that have been answered in the survey. */
+  fun trackAnsweredQuestions(questionName: SurveyQuestionName) {
+    answeredQuestions.add(questionName)
+  }
+
+  /** Returns whether the survey progress has reached the threshold for partial completion. */
+  fun hasReachedPartialCompletionThreshold(): Boolean {
```

Comment:
Is "partially completed" (up to the threshold) a subset of "completed" (i.e. would a survey reaching that threshold be also considered complete in a similar manner to isCompletion above), or is it a distinct category?

It's a bit ambiguous, so I would suggest something like hasCompletedAllMandatoryQuestions or something that actually includes the definition of "partial completion" (so that you don't need to use that jargon).

Reply:
> Is "partially completed" (up to the threshold) a subset of "completed" (i.e. would a survey reaching that threshold be also considered complete in a similar manner to isCompletion above), or is it a distinct category?

Your question is not very clear, but I'll try to answer it:
The survey can have 3 completion states: Fully completed, partially completed or not completed.
It is only fully completed iff all the mandatory questions have been answered and the optional question has been answered as well. It is Partially complete if the mandatory questions have been answered but the optional question has not. If the Mandatory question threshhold hasn't been reached, we consider the survey abandoned.

isCompletion(updated name) is only ever true if the optional question has been answered.

> It's a bit ambiguous, so I would suggest something like hasCompletedAllMandatoryQuestions or something that actually includes the definition of "partial completion" (so that you don't need to use that jargon).

I have updated the function name to `hasAnsweredAllMandatoryQuestions`